modules/reindex/src/test/java/org/elasticsearch/index/reindex/ReindexSourceTargetValidationTests.java

```diff
@@ -61,6 +63,8 @@
     private static final AutoCreateIndex AUTO_CREATE_INDEX = new AutoCreateIndex(Settings.EMPTY,
             new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS), INDEX_NAME_EXPRESSION_RESOLVER);
 
+    private final BytesReference matchAll = new BytesArray(new MatchAllQueryBuilder().toString());
```

Comment:
I would prefer `new BytesArray("{ \"foo\" : \"bar\" }"`, makes it more explicit what we rely on (JSON and not query-validity).